org.jacoco.core.test/src/org/jacoco/core/internal/analysis/filter/KotlinComposeFilterTest.java

```diff
@@ -0,0 +1,255 @@
+	 *   if ($dirty and 0b0011 != 0b0010 || !$composer.skipping) {
+	 *     if (isTraceInProgress()) {
+	 *       traceEventStart(...)
+	 *     }
+	 *
+	 *     if (x < 0) {
+	 *       if (isTraceInProgress()) {
+	 *         traceEventEnd()
+	 *       }
+	 *       $composer.endRestartGroup()?.updateScope { ... }
+	 *       return
+	 *     }
+	 *
+	 *     example(x, $composer, 0)
+	 *
+	 *     if (isTraceInProgress()) {
+	 *       traceEventEnd()
+	 *     }
+	 *   } else {
+	 *     $composer.skipToGroupEnd()
+	 *   }
+	 *   $composer.endRestartGroup()?.updateScope { ... }
+	 * }
+	 * </pre>
+	 */
+	@Test
+	public void should_filter() {
+		context.classAnnotations
+				.add(KotlinGeneratedFilter.KOTLIN_METADATA_DESC);
+		final MethodNode m = new MethodNode(InstrSupport.ASM_API_VERSION, 0,
+				"f", "(Landroidx/compose/runtime/Composer;I)V", null, null);
+		m.visitAnnotation("Landroidx/compose/runtime/Composable;", false);
+
+		final Range range1 = new Range();
+		final Range range2 = new Range();
+		final Range range3 = new Range();
+		final Range range4 = new Range();
+		final Range range5 = new Range();
+		final Range range6 = new Range();
+		final Range range7 = new Range();
+
+		m.visitVarInsn(ALOAD, 1);
+		range1.fromInclusive = m.instructions.getLast();
+		m.visitLdcInsn(new Integer(-974630231));
+		m.visitMethodInsn(INVOKEINTERFACE, "androidx/compose/runtime/Composer",
+				"startRestartGroup", "(I)Landroidx/compose/runtime/Composer;",
+				true);
+		m.visitVarInsn(ASTORE, 1);
+		m.visitVarInsn(ILOAD, 2);
+		m.visitVarInsn(ISTORE, 3);
+		Label label1 = new Label();
+		m.visitLabel(label1);
+		m.visitVarInsn(ILOAD, 2);
+		m.visitIntInsn(BIPUSH, 14);
+		m.visitInsn(IAND);
+		Label label2 = new Label();
+		m.visitJumpInsn(IFNE, label2);
+		m.visitVarInsn(ILOAD, 3);
+		m.visitVarInsn(ALOAD, 1);
+		m.visitVarInsn(ILOAD, 0);
+		m.visitMethodInsn(INVOKEINTERFACE, "androidx/compose/runtime/Composer",
+				"changed", "(I)Z", true);
+		Label label3 = new Label();
+		m.visitJumpInsn(IFEQ, label3);
+		m.visitInsn(ICONST_4);
+		Label label4 = new Label();
+		m.visitJumpInsn(GOTO, label4);
+		m.visitLabel(label3);
+		m.visitInsn(ICONST_2);
+		m.visitLabel(label4);
+		m.visitInsn(IOR);
+		m.visitVarInsn(ISTORE, 3);
+		m.visitLabel(label2);
+		m.visitVarInsn(ILOAD, 3);
+		m.visitIntInsn(BIPUSH, 11);
+		m.visitInsn(IAND);
+		m.visitInsn(ICONST_2);
+		Label label5 = new Label();
+		m.visitJumpInsn(IF_ICMPNE, label5);
+		m.visitVarInsn(ALOAD, 1);
+		m.visitMethodInsn(INVOKEINTERFACE, "androidx/compose/runtime/Composer",
+				"getSkipping", "()Z", true);
+		Label label6 = new Label();
+		m.visitJumpInsn(IFNE, label6);
+		range1.toInclusive = m.instructions.getLast();
+
+		m.visitLabel(label5);
+		m.visitMethodInsn(INVOKESTATIC, "androidx/compose/runtime/ComposerKt",
+				"isTraceInProgress", "()Z", false);
+		Label label7 = new Label();
+		m.visitJumpInsn(IFEQ, label7);
+		range3.fromInclusive = m.instructions.getLast();
+		m.visitLdcInsn(new Integer(-974630231));
```

Comment:
```suggestion
		m.visitLdcInsn(Integer.valueOf(-974630231));
```
Avoid deprecation warning on newer JDKs.